qutebrowser/mainwindow/tabbedbrowser.py

```diff
@@ -487,12 +489,21 @@ def tabopen(self, url=None, background=None, related=True, idx=None, *,
             tab.resize(self.widget.currentWidget().size())
             self.widget.tab_index_changed.emit(self.widget.currentIndex(),
                                                self.widget.count())
+            # Refocus webview in case we lost it by spawning a bg tab
+            self.widget.currentWidget().setFocus()
         else:
             self.widget.setCurrentWidget(tab)
             # WORKAROUND for https://bugreports.qt.io/browse/QTBUG-68076
             # Still seems to be needed with Qt 5.11.1
             tab.setFocus()
 
+        mode = modeman.instance(self._win_id).mode
+        if mode in [usertypes.KeyMode.command, usertypes.KeyMode.prompt,
+                    usertypes.KeyMode.yesno]:
+            # If we were in a command prompt, restore old focus
+            # The above commands need to be run to switch tabs
+            prev_focus.setFocus()
```

Comment:
This seems weird. Maybe we should just always focus the current webview when one of those modes is left?